web/components/MsoaBoundaries.svelte

```diff
@@ -0,0 +1,175 @@
+<script>
+  import { getContext, onMount, onDestroy } from "svelte";
+  import bbox from "@turf/bbox";
+  import chroma from "chroma-js";
+
+  const { getMap, setCamera } = getContext("map");
+  let map = getMap();
+
+  export let msoas;
+  export let hoveredMsoa;
+
+  let colorBy = "households";
+  let limits = [];
+  let colorScale = [];
+
+  let source = "msoas";
+  let layer = "msoas-polygons";
+
+  let hoverId;
+  function unhover() {
+    if (hoverId != null) {
+      map.setFeatureState({ source, id: hoverId }, { hover: false });
+    }
+  }
+
+  onMount(() => {
+    let gj = {
+      type: "FeatureCollection",
+      features: Object.values(msoas),
+    };
+
+    if (setCamera) {
+      map.fitBounds(bbox(gj), {
+        padding: 20,
+        animate: false,
+      });
+    }
+
+    // TODO If we pass the MSOA ID as feature.id, it gets dropped?
+    map.addSource(source, { type: "geojson", data: gj, generateId: true });
+
+    // TODO Just fill-outline-color
+    map.addLayer({
+      id: "msoas-lines",
+      source,
+      type: "line",
+      paint: {
+        "line-color": "black",
+        "line-width": 3,
+      },
+    });
+
+    setLayer();
+
+    map.on("mousemove", layer, (e) => {
+      if (e.features.length > 0 && hoverId != e.features[0].id) {
+        unhover();
+        hoveredMsoa = e.features[0].properties.id;
+        hoverId = e.features[0].id;
+        map.setFeatureState({ source, id: hoverId }, { hover: true });
+      }
+    });
+    map.on("mouseleave", layer, () => {
+      unhover();
+      hoveredMsoa = null;
+      hoverId = null;
+    });
+  });
+
+  function setLayer() {
+    if (map.getLayer(layer)) {
+      map.removeLayer(layer);
+    }
+
+    // Get the numeric data we're displaying
+    let data = Object.values(msoas).reduce((agg, msoa) => {
+      agg.push(msoa.properties[colorBy]);
+      return agg;
+    }, []);
+    // chroma equidistant scale
```

Comment:
This is for numeric/sequential data only, and we'll definitely want to play around with the settings here